### PR TITLE
Add K-LMS scheduler from k-diffusion

### PR DESCRIPTION
**Pull request by anton-l**
This PR adds the K-LMS sampler from [k-diffusion](https://github.com/crowsonkb/k-diffusion/blob/master/k_diffusion/sampling.py) by Katherine Crowson.

At the moment it only supports discrete beta-schedules (specifically the one for Stable Diffusion) but it will be extended to support continuous sigma-schedules in a follow-up PR.

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py [hunk 1]
```diff
@@ -118,7 +125,10 @@ def __call__(
                 noise_pred = noise_pred_uncond + guidance_scale * (noise_pred_text - noise_pred_uncond)
 
             # compute the previous noisy sample x_t -> x_t-1
-            latents = self.scheduler.step(noise_pred, t, latents, **extra_step_kwargs)["prev_sample"]
+            if isinstance(self.scheduler, LmsDiscreteScheduler):
+                latents = self.scheduler.step(noise_pred, i, latents, **extra_step_kwargs)["prev_sample"]
+            else:
+                latents = self.scheduler.step(noise_pred, t, latents, **extra_step_kwargs)["prev_sample"]
```
Comment (anton-l) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py:131]: This can be made cleaner by removing the branch with `t`->`i`, but for now I'm not touching the timesteps from k-lms for ease of debugging.

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py [hunk 2]
```diff
@@ -105,9 +105,16 @@ def __call__(
...
+        self.scheduler.set_timesteps(num_inference_steps)
+        if isinstance(self.scheduler, LmsDiscreteScheduler):
+            latents = latents * self.scheduler.sigmas[0]
+
+        for i, t in tqdm(enumerate(self.scheduler.timesteps)):
             # expand the latents if we are doing classifier free guidance
             latent_model_input = torch.cat([latents] * 2) if do_classifier_free_guidance else latents
+            if isinstance(self.scheduler, LmsDiscreteScheduler):
+                sigma = self.scheduler.sigmas[i]
+                latent_model_input = latent_model_input / ((sigma**2 + 1) ** 0.5)
```
Comment (anton-l) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py:117]: We'll probably need a mechanism to specify custom coefficients for inputs in the future, since the Karras scheduler needs it too.

File: src/diffusers/__init__.py [hunk 3]
```diff
@@ -27,11 +27,17 @@
     SchedulerMixin,
     ScoreSdeVeScheduler,
 )
+
+
+if is_scipy_available():
+    from .schedulers import LmsDiscreteScheduler
```
Comment (patrickvonplaten) [inline src/diffusers/__init__.py:33]: What does lms stand for? Maybe `LMS` for consistency?

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py [hunk 4]
```diff
@@ -118,7 +125,10 @@ def __call__(
                 noise_pred = noise_pred_uncond + guidance_scale * (noise_pred_text - noise_pred_uncond)
 
             # compute the previous noisy sample x_t -> x_t-1
-            latents = self.scheduler.step(noise_pred, t, latents, **extra_step_kwargs)["prev_sample"]
+            if isinstance(self.scheduler, LmsDiscreteScheduler):
+                latents = self.scheduler.step(noise_pred, i, latents, **extra_step_kwargs)["prev_sample"]
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py:129]: shouldn't we maybe pass the sigma here?

**Review by patrickvonplaten — COMMENTED** [review]
Just wondering about the `i` vs. sigma API since in karras_ve we pass a sigma: 
https://github.com/huggingface/diffusers/blob/9070c394aa69810a51293b6e18de940f14dd892f/src/diffusers/schedulers/scheduling_karras_ve.py#L102

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py (hunk 4 above)
Comment (patrickvonplaten) [reply src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py:129]: Ok we need it anyway - good to leave as is for me!

**Review by patrickvonplaten — APPROVED** [review]

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py (hunk 4 above)
Comment (anton-l) [reply src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py:129]: The `i` is needed inside the `step` for other calculations too, and we can't reverse the sigma into `i`, so we'll need it like this until a more full refactor.

**Comment by rossli0102** [comment]
prompt = ["a logo of Knicks and [championship](javascript:;) "]

height = 512                        # default height of Stable Diffusion
width = 512                         # default width of Stable Diffusion

num_inference_steps = 100            # Number of denoising steps

guidance_scale = 7.5                # Scale for classifier-free guidance

generator = torch.manual_seed(32)   # Seed generator to create the inital latent noise

batch_size = 1
